File: 30_days_of_code/10-binary-numbers/py/binary-numbers.py
N = int(raw_input().strip())
count = max = 0

i = 1
while i < N:
    if (N & i) == i:
        count = count + 1
    else:
        if count > max:
            max = count
        count = 0
    i = i*2

if count > max: max = count
print max

File: 30_days_of_code/10-binary-numbers/cpp/binary-numbers.cpp
#include <iostream>
using namespace std;


int main() {
  int n;
  cin >> n;
  int count =0, max =0;
  for(int x = 1; x < n; x *= 2) {
    if((n & x) == x) {count++;}
    else { (count > max) && (max = count); count =0;}
  }
  (count > max) && (max = count);
  cout << max << endl;
  return 0;
}
